ui/src/reducers/resource.js

```diff
@@ -0,0 +1,61 @@
+const defaultState = {
+  isFetching: false,
+  items: [],
+  fetchError: false,
+  isCreating: false,
+  createError: false
+}
+
+const resourceReducer = (actions) => (state = defaultState, action) => {
+  if (actions.listRequest) {
+    switch (action.type) {
+      case actions.listRequest:
+        return {
+          ...state,
+          isFetching: true,
+          fetchError: false
+        }
+      case actions.listSuccess:
+        return {
+          ...state,
+          isFetching: false,
+          items: action.result,
+          fetchError: false
+        }
+      case actions.listFailure:
+        return {
+          ...state,
+          isFetching: false,
+          fetchError: action.error
+        }
+      default:
+    }
+  }
+  if (actions.createRequest) {
```

Comment:
This `if` checks if there are actions defined for create requests. If so it will check if such an action was dispatched. If there was a `else if`, reducers for create actions would not be called as every resource also has a list actions.

Reply:
Ok, understand. Wondering if we really need the outer `if`. `actions.createRequest` could be undefined but that would be ok because `action.type` should never be undefined. WDYT?

Reply:
yes, ok, that makes sense, will fix it